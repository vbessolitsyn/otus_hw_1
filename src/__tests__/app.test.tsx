import { render } from '@testing-library/react';

import App from '../app';

test('renders app component', () => {
  const { container, getByRole } = render(<App />);

  expect(getByRole('main')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('catch evaluate result', () => {
  const { container } = render(<App />);
  container.getElementsByClassName('input')[0].innerHTML = '123 + 321';
  container.getElementsByClassName('result')[0].innerHTML = 'Результат: 444';
});

test('catch evaluate exceptions', () => {
  const { container } = render(<App />);
  container.getElementsByClassName('input')[0].innerHTML = 'error content';
  container.getElementsByClassName('result')[0].innerHTML =
    'Результат: Expression error content is not valid';
});
