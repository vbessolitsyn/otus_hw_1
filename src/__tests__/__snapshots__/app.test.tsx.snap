// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component 1`] = `
<div>
  <div
    role="main"
  >
    <h1
      class="header"
    >
       Задание 1: калькулятор 
    </h1>
    <p
      class="main"
    >
      Введите выражение и нажмите [Enter]:
      <br />
      <br />
      <input
        class="input"
        size="100"
        spellcheck="false"
        type="text"
      />
      <br />
      <br />
      Результат: 
      <span
        class="result"
      />
    </p>
  </div>
</div>
`;
